test/seam/connect/seam-http-request.test.ts

```diff
@@ -0,0 +1,36 @@
+import test from 'ava'
+import { getTestServer } from 'fixtures/seam/connect/api.js'
+
+import { SeamHttp } from '@seamapi/http/connect'
+
+import {
+  type ResponseFromSeamHttpRequest,
+  SeamHttpRequest,
+} from 'lib/seam/connect/seam-http-request.js'
+
+test('serializes array params when undefined', async (t) => {
+  const { seed, endpoint } = await getTestServer(t)
+  const seam = SeamHttp.fromApiKey(seed.seam_apikey1_token, { endpoint })
+
+  const deviceRequest = seam.devices.get({ device_id: seed.august_device_1 })
+
+  t.true(deviceRequest instanceof SeamHttpRequest)
+  t.is(deviceRequest.url, '/devices/get')
+  t.deepEqual(deviceRequest.data, {
+    device_id: seed.august_device_1,
+  })
+  t.is(deviceRequest.resourceKey, 'device')
+  const device = await deviceRequest
+  t.is(device.workspace_id, seed.seed_workspace_1)
+  t.is(device.device_id, seed.august_device_1)
+
+  // Ensure that the type of the response is correct.
+  type Expected = ResponseFromSeamHttpRequest<typeof deviceRequest>
```

Comment:
Do we need `ResponseFromSeamHttpRequest` or can we just use `DevicesGetResponse`? https://github.com/seamapi/javascript-http/pull/67/files#diff-dbc791a08a86b8bf7b54793b2bbed5d188cb1348e81aeba26f877fd2ff2ab202R260-R262

Reply:
I want to ensure that type inference keeps working so I'd rather use the request, I've moved the type into the test though in my next commit as it's unused elsewhere.